### ISSUE TITLE
Bridge interfaces in state DOWN

### ISSUE THREAD

**Issue by pfl** [systemd/systemd#10441]
Systemd upstream at commit 17fd78cb864421a23171281809370822f1de8e96 compiled on Debian testing configured to create a bridge interface and assign an IPv6 address to it. The idea is to create a bridge and watch systemd-networkd configure the link before any virtual machines are involved, especially to verify those downstream Router Advertisements containing prefixes assigned by DHCPv6 on another link.

/etc/systemd/network/virbrs1.netdev:
[Match]

[NetDev]
Description=Systemd-networkd static bridge network #1 for virtual machines
Name=virbrs1
Kind=bridge

/etc/systemd/network/virbrs1.network:
[Match]
Name=virbrs1

[Network]
Description=Systemd-networkd virbrs1 configuration
ConfigureWithoutCarrier=true
Address=fd66:6c79:6b74:7301::1/64

After restart, systemd-networkd detects the intended configuration, but does not manage to bring up the link. The relevant log output looks like:
virbrs1: loaded bridge
virbrs1: Creating
virbrs1: Flags change: +MULTICAST +BROADCAST
virbrs1: Link 21 added
virbrs1: udev initialized link
virbrs1: netdev has index 21
virbrs1: netdev ready
virbrs1: Saved original MTU: 1500

'ip link' gives the following information:
21: virbrs1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether 1e:8a:1c:94:c1:71 brd ff:ff:ff:ff:ff:ff

If the bridge link is created before starting systemd-networkd with 'brctl addbr virbrs1', it has all the necessary flags set, and the IPv6 address can be assigned. The difference in flags is:
virbrs1: Flags change: +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST

'ip link' gives the following information:
21: virbrs1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether 1e:8a:1c:94:c1:71 brd ff:ff:ff:ff:ff:ff

'ip address' shows:
21: virbrs1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether 1e:8a:1c:94:c1:71 brd ff:ff:ff:ff:ff:ff
    inet6 fd66:6c79:6b74:1::/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::1c8a:1cff:fe94:c171/64 scope link 
       valid_lft forever preferred_lft forever

Apparently ConfigureWithoutCarrier is not checked in all places where link/IP configuration needs to be accounted for? Or what is the intended behavior here?

**Comment by pfl** [comment]
More background to this: some services need to be runing for the virtual machines to work properly. The services don't support to be configured with an IP address that is not present when the daemon starts, and since there are no dependencies on specific interfaces being enabled, the services cannot be automatically started when the interface goes up. The only dependency I can think of is to wait for network online, but if the interface is not up, the services will not start. And if one waits for the interface to be enabled, the boottime goes through the roof and/or the system does not start at all.

So this could be solved by adding a switch to .netdev files, say "BringUpInterface=yes". Is this the way to proceed? Right now the ugly fix is to do /etc/systemd/system/systemd-networkd.service.d/bridge.conf like so:
[Service]
ExecStartPre=/bin/sh -c "/sbin/brctl addbr virbrs1 || /bin/true"

**Comment by cbpeckles** [comment]
@pfl I was able to work around a similar problem by using a dummy interface and binding that to the bridge along with any other devices.  My use case sounds similar to yours in that my bridge was intended to bind to interfaces that are not yet created at boot.  This resulted in the bridge coming up and other services depending on network-online to start.  YMMV.  This has left me happier than modifying ExecStartPre however the problem with systemd apparently ignoring ConfigureWithoutCarrier still needs some attention.

**Comment by yuwata** [comment]
Duplicate of #9252.
